platform/jvm/capture/src/main/kotlin/io/bitdrift/capture/reports/FatalIssueReporter.kt

```diff
@@ -28,31 +41,155 @@ import kotlin.time.measureTime
  */
 internal class FatalIssueReporter(
     private val mainThreadHandler: MainThreadHandler = MainThreadHandler(),
-) {
-    private val appContext = APP_CONTEXT
+    private val latestAppExitInfoProvider: ILatestAppExitInfoProvider = LatestAppExitInfoProvider(),
+    private val captureUncaughtExceptionHandler: ICaptureUncaughtExceptionHandler = CaptureUncaughtExceptionHandler(),
+) : IFatalIssueReporter {
+    private var fatalIssueReporterStatus: FatalIssueReporterStatus = buildDefaultReporterStatus()
+    private val appContext by lazy { APP_CONTEXT }
+    private val sdkDirectory by lazy { SdkDirectory.getPath(appContext) }
+    private val destinationDirectory: File by lazy {
+        File(sdkDirectory, DESTINATION_FILE_PATH).apply { if (!exists()) mkdirs() }
+    }
+    private val fatalIssueReporterProcessor: FatalIssueReporterProcessor by lazy {
+        FatalIssueReporterProcessor(FatalIssueReporterStorage(destinationDirectory))
+    }
+    private val activityManager: ActivityManager by lazy {
+        appContext.getSystemService(Context.ACTIVITY_SERVICE) as ActivityManager
+    }
+
+    override fun initialize(fatalIssueMechanism: FatalIssueMechanism) {
+        if (fatalIssueReporterStatus.state is FatalIssueReporterState.NotInitialized) {
+            if (fatalIssueMechanism == FatalIssueMechanism.CUSTOM_CONFIG) {
+                fatalIssueReporterStatus = processReportFilesFromCustomConfig()
+            } else if (fatalIssueMechanism == FatalIssueMechanism.BUILT_IN) {
+                fatalIssueReporterStatus = initializeBuiltInReporting()
+            }
+        } else {
+            Log.w("capture", "Fatal issue reporting already being initialized")
+        }
+    }
+
+    override fun fetchStatus(): FatalIssueReporterStatus = fatalIssueReporterStatus
```

Comment:
the use of the `fetch` verb here makes me think this is an I/O operation or something of the sort